main/src/library/File.flix

```diff
@@ -364,6 +364,43 @@ namespace File {
             }
         }
 
+    ///
+    /// Returns an iterator of the bytes in the given `file` in chunks of size `chunkSize`.
+    ///
+    pub def readChunks(f: String, chunkSize: Int32): Iterator[Result[Array[Int8], String]] & Impure =
+        try {
+            import new java.io.File(String) as newFile;
+            import new java.io.FileInputStream(##java.io.File) as newInputStream;
+            import java.io.FileInputStream.read(Array[Int8]);
+
+            let javaFile = newFile(f);
+            let stream = newInputStream(javaFile);
+            let bytes = ref [0i8; chunkSize];
+            let numberRead = ref read(stream, deref bytes);
+
+            let done = () -> deref numberRead == -1;
+            let next = () -> {
+                try {
+                    let chunk = deref bytes;
+                    bytes := [0i8; chunkSize];
+                    let accRead = deref numberRead;
+                    numberRead := read(stream, deref bytes);
+                    Ok(readBytesWithHelper(chunk, accRead, chunkSize))
+                } catch {
+                    case ex: ##java.io.IOException => {
+                        bytes := null as Array[Int8];
```

Comment:
Setting to `null` was our way of signaling "done" before (to end the iterator after an error). But since the `done` function is checking `numberRead` instead, we should use `numberRead` to signal the end in this case.

Reply:
Ah, yes, you are right

Reply:
I cannot seem to get it to work. Do you know why, it is not possible to call `Object.isNull(null as Int32)`, but it works for `Object.isNull(null as String)`.

Reply:
`Int32` is a Java primitive, which cannot be `null` (only objects can be null). Instead of setting/checking for null, we can keep the `-1` test and set `numberRead` to `-1`.

Reply:
Yes, good idea